File: leetcode/python/1497.check-if-array-pairs-are-divisible-by-k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        rems = [0] * k

        for n in arr:
            rems[((n % k) + k) % k] += 1
        
        if rems[0] & 1 != 0:
            return False
        
        for i in range(1, 1 + k // 2):
            if rems[i] != rems[k-i]:
                return False
        
        return True

File: leetcode/cpp/1497.check-if-array-pairs-are-divisible-by-k.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        vector<int> rems(k, 0);

        for (const int n: arr) {
            rems[((n % k) + k) % k]++;
        }

        if (rems[0] & 1 == 1) {
            return false;
        }

        for (int i = 1; i < 1 + k/2; i++) {
            if (rems[i] != rems[k-i]) {
                return false;
            }
        }
        return true;
    }
};

